### PR TITLE
Fix for issue 121

### PR DESCRIPTION
**Pull request by mbertrand**
@mgrauer This should fix the issue you reported (https://github.com/Kitware/minerva/issues/121)

Steps to reproduce issue:
- Add a new WMS source
- Add a dataset from the source
- Add another WMS source
- Add a dataset from the second WMS source

### CODE REVIEW

**Comment by mgrauer** [comment]
Thanks @mbertrand .  This fixes the 404, but if you add a layer from a wms source, then add a layer from that same source, it will add multiple layers (2 the second time, 3 the third time).  Every time you create a new widget, you attach another event listener to the form, which is why we have the multiple adds.  

Your fix pointed me in the right direction, though, I've got a fix pushed.

File: web_external/js/views/body/SourcePanel.js [hunk 1]
```diff
@@ -20,17 +20,13 @@ minerva.views.SourcePanel = minerva.View.extend({
...
-                el: $('#g-dialog-container'),
-                source: wmsSource,
-                collection: this.datasetCollection,
-                parentView: this
-            });
-            this.wmsLayersListWidget.render();
-        } else {
-            this.wmsLayersListWidget.setCurrentSource(wmsSource);
-        }
+        this.wmsLayersListWidget = new minerva.views.WmsLayersListWidget({
```
Comment (mbertrand) [inline web_external/js/views/body/SourcePanel.js]: @mgrauer, if I add this snippet just before line 23 then the dupe issue goes away:
```
        if (this.wmsLayersListWidget) {
            this.wmsLayersListWidget.destroy();
        }
```
But maybe this isn't as efficient as reusing the existing widget.  What do you think?

**Comment by mgrauer** [comment]
That is good to know @mbertrand , and it makes sense, thanks.

I'd rather go with the solution in #128 to keep consistent with best practices in Girder, especially as I think using the `girderModal` will be needed to set up client side testing correctly (or at least so we can tap into the existing testing infrastructure and institutional knowledge) once we finally get to testing there.

**Comment by mbertrand** [comment]
Thanks @mgrauer, should I go ahead and close this PR?

**Comment by mgrauer** [comment]
Yes, I'll do it, just wanted to complete the discussion first.

**Pull request closed without merging**